Sources/TripKit/helpers/TKUserProfileHelper.swift

```diff
@@ -132,5 +133,40 @@ public class TKUserProfileHelper: NSObject {
     }
   }
   
+  // MARK: - Mode by mode, Mode picker
+  
+  /// update picked modes provided with all reference modes to determine disabled modes
+  @objc public class func update(pickedModes: Set<TKModeInfo>, allModes: Set<TKModeInfo>) {
+    var modes = disabledSharedVehicleModes
+    
+    let pickedIds = pickedModes.compactMap { try? JSONEncoder().encode($0)  }
+    let allIds = allModes.compactMap { try? JSONEncoder().encode($0)  }
+    
+    let toDisable = allIds.filter {
+      !pickedIds.contains($0)
+    }
+    
+    pickedIds.forEach { mode in
+      if modes.contains(mode) {
+        modes = modes.filter { mode != $0 }
+      }
+    }
+    
+    toDisable.forEach {
+      if !modes.contains($0) {
+        modes.append($0)
+      }
+    }
```

Comment:
This is the union of `toDisable` and `modes`. So it can be just:

```suggestion
    modes.formUnion(toDisable)
```